### ISSUE TITLE
GestureRecognizerCollection not releasing capture

### ISSUE THREAD

**Issue by aguahombre** [AvaloniaUI/Avalonia#3133]
v0.9.0 preview1
The GestureRecognizerCollection doesn't release ```_pointerGrabs``` in ```HandlePointerReleased``` so a control with multiple gesture recognizers only receives gestures from the first gesture that captures the pointer.

**Comment by kekekeks** [comment]
PointerReleased event should be immediately followed with PointerCaptureLost event.

**Comment by aguahombre** [comment]
Interesting, I have a breakpoint on PointerCaptureLost in my gesture recognizer but it never gets hits it.

**Comment by aguahombre** [comment]
A quick look at the source shows MouseUp handler calls ```_pointer.Capture(null)``` and this won't generate a PointerCaptureLost event.

**Comment by aguahombre** [comment]
There is another problem with the gestures recognizers collection. 
I will try to explain with a scenario example:

The gestures recognizers collection contains two gesture recognizers
- one is looking for single finger touch gestures
- the second is looking for two finger touch gestures

Both recognizers will receive a ```PointerPressed``` event when the first finger touch occurs, both store the pointer information (touch ID etc).
The finger is moved triggering ```PointerMoved``` events for both recognizers.
The first recognizer now decides this is a single finger gesture and captures the input via ```IGestureRecognizerActionsDispatcher```.
From this time forwards only the first recognizer receives pointer (touch) events for the touch ID.

The finger is removed and the first recognizer gets a ```PointerReleased``` event.
The second recognizer never gets a notification to discard the touch ID.

Now another single finger down occurs and both recognizers receive a ```PointerPressed``` event with a new touch ID.
The second gesture recognizer now thinks there is a two finger touch in progress.
The first recognizer thinks there is a single touch in progress.

**Comment by aguahombre** [comment]
Maybe GestureRecognizerCollection could be modified to do something like this?

```
void IGestureRecognizerActionsDispatcher.Capture(IPointer pointer, IGestureRecognizer recognizer)
{
	pointer.Capture(_inputElement);
	_pointerGrabs[pointer] = recognizer;
	
	foreach (var r in _recognizers)
	{
		if (r != recognizer)
		{
		     r.PointerCapturedElsewhere();
		}
	}
}
```

**Comment by kekekeks** [comment]
It should trigger PointerCaptureLost for the second recognizer, I think

**Comment by aguahombre** [comment]
Yes, I did think about that but thought maybe it was semantically incorrect as the second recognizer had never captured the pointer.

**Comment by Mikolaytis** [comment]
@kekekeks is this fixed by #3594? let's close this then
